### PR TITLE
Refactor `analytics`

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR
- refactors the analytics codebase to more closely match the structure of ingestion-server
- refactors the ingestion-server API to reflect improvements from analytics
- sets up automatic swagger and ReDoc generation
- fixes an ORIGIN bug (should not contain path) and completes a todo task
- improves code quality

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
It looks like this might need the latest changes from `main` (via #438)

**Comment by dhruvkb** [comment]
Number of changed files went from 179 to 36 😆.

File: analytics/analytics/docs/schemas/requests.py
Comment (AetherUnbound) [inline analytics/analytics/docs/schemas/requests.py:3]: Very cool! I've been wanting to use pydantic :rocket:

File: analytics/analytics/server.py
Comment (AetherUnbound) [inline analytics/analytics/server.py:81]: A lot of these docstrings start with `"""create`, is it alright to remove that bit?

**Review by AetherUnbound — APPROVED** [review]
Just a couple small notes, nice work!

File: analytics/analytics/server.py
Comment (dhruvkb) [reply analytics/analytics/server.py:81]: These are used to give operation IDs to the endpoints (shows up in the ReDoc as headings, which would otherwise default to `on_post <POST>`)

File: analytics/analytics/docs/schemas/requests.py
Comment (dhruvkb) [reply analytics/analytics/docs/schemas/requests.py:3]: Yeah, Pydantic is indeed very cool! It basically removed the need for most of the input validation inside the `on_get` and `on_post` handlers.

File: analytics/analytics/server.py
Comment (AetherUnbound) [reply analytics/analytics/server.py:81]: Got it! That makes sense, I just wasn't aware :slightly_smiling_face: My naive assumption was copypasta leftovers :grin:

File: analytics/analytics/docs/redoc.html
Comment (sarayourfriend) [inline analytics/analytics/docs/redoc.html:15]: Whoops! Looks like an accidental edit here adding unnecessary extra curly braces.

File: analytics/gunicorn.conf.py
Comment (sarayourfriend) [inline analytics/gunicorn.conf.py:6]: Should this be configured by environment?

**Review by sarayourfriend — COMMENTED** [review]
This looks amazing @dhruvkb! I'm excited to see some pydantic, it's such a cool library.

What's the best way to test this PR?

File: analytics/analytics/docs/redoc.html
Comment (dhruvkb) [reply analytics/analytics/docs/redoc.html:15]: Spectree (the Python plugin we're using to generate ReDoc) reads the file as a string and uses the `.format` function to populate the Swagger URL. To prevent Python from trying to interpolate the contents of `{ }`, we must [double the braces](https://stackoverflow.com/a/5466478) so that Python treats them as literals.

File: analytics/gunicorn.conf.py
Comment (dhruvkb) [reply analytics/gunicorn.conf.py:6]: It could be. But currently all Gunicorn configs are set to be log-level debug.

**Comment by dhruvkb** [comment]
@sarayourfriend you could test the endpoints by making `POST` requests using Postman. I have no idea how one could test the reporting (yet).

**Review by zackkrida — APPROVED** [review]
Excellent, excellent work. To @dhruvkb's suggestion I would also recommend [Insomnia](https://github.com/Kong/insomnia) as an open-source alternative to Postman.

**Comment by AetherUnbound** [comment]
I've used [Hopscotch.io](hoppscotch.io/) (formerly Postwoman) in the past :slightly_smiling_face:

**Comment by zackkrida** [comment]
awh, bummer, i loved the postwoman name and brand haha

**Comment by AetherUnbound** [comment]
Ha, same! The tool still rocks though

File: analytics/analytics/docs/redoc.html
Comment (sarayourfriend) [reply analytics/analytics/docs/redoc.html:15]: 😮 That's interesting. Should we really use `html` as the file extension then? That's pretty surprising behavior for an html file that doesn't really have any other indiciation that it's actually a template string... Maybe that's just me though.

**Review by krysal — COMMENTED** [review]
This looks like a lot of improvements! I probably need to do additional step(s) but when I visit http://localhost:8090/ I only got a response with `{"status": "200 OK"}`. The README says:

> After starting the server, you can view the documentation by visiting the root path (e.g. localhost:8090/). You may have to tweak `docs/redoc.html` for this to work on your local machine.

What should I do to that file to see the documentation locally?

**Comment by dhruvkb** [comment]
@krysal you have to visit `/doc/redoc`, not the root path. Sorry for the wrong instructions.

**Comment by dhruvkb** [comment]
This has two approvals, so I'll merge it to unblock @AetherUnbound's PRs that target this branch. We can address any changes in a separate small PR.

**Pull request merged**
